app/controllers/users_controller.rb

```diff
@@ -0,0 +1,38 @@
+class UsersController < ApplicationController
```

Comment:
I'm confused why you have a users controller/model/view that mimics the Markets?  I've made comments elsewhere on how to refactor out the users controller contents into the markets controller contents.